cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/cruisecontrol/CruiseControlClientTest.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.operator.assembly.cruisecontrol;
+
+import io.vertx.core.Vertx;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockserver.integration.ClientAndServer;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.CoreMatchers.is;
+
+@ExtendWith(VertxExtension.class)
+public class CruiseControlClientTest {
+
+    private static final int PORT = 1080;
+    private static final String HOST = "localhost";
+
+    private static ClientAndServer ccServer;
+
+    @BeforeAll
+    public static void setupServer() throws IOException, URISyntaxException {
+        ccServer = MockCruiseControl.getCCServer(PORT, 2);
+    }
+
+    @AfterAll
+    public static void stopServer() {
+        ccServer.stop();
+    }
+
+    @Test
+    public void testGetCCState(Vertx vertx, VertxTestContext context) {
+
+        CruiseControlApi client = new CruiseControlApiImpl(vertx);
+
+        client.getCruiseControlState(HOST, PORT, false).setHandler(result -> {
+            if (result.succeeded()) {
+                context.verify(() -> assertThat(
+                        result.result().getJson().getJsonObject("ExecutorState").getString("state"),
+                        is("NO_TASK_IN_PROGRESS")));
+                context.completeNow();
+            } else {
+                context.failNow(result.cause());
+            }
+        });
+
+    }
+
+    @Test
+    public void testCCRebalance(Vertx vertx, VertxTestContext context) {
+
+        RebalanceOptions rbOptions = new RebalanceOptions.RebalanceOptionsBuilder().build();
+
+        CruiseControlApi client = new CruiseControlApiImpl(vertx);
+
+        client.rebalance(HOST, PORT, rbOptions).setHandler(result -> {
+            if (result.succeeded()) {
```

Comment:
maybe we can simplify using this : https://github.com/strimzi/strimzi-kafka-operator/commit/e54647be337afebac069cb87f26f6eb2b84000fb

See the usage of `context.succeeding` and then just one `verify`.

Reply:
Oh good idea, I will have a look

Reply:
Done, switched all the tests over to `contex.succeeding`